game-service/src/main/java/org/libertybikes/game/round/service/GameRoundService.java

```diff
@@ -48,14 +48,17 @@ public String createRound() {
 
     @POST
     public GameRound createRoundById(@QueryParam("gameId") String gameId) {
+        if (allRounds.containsKey(gameId)) {
```

Comment:
This fixes double creating game rounds, which messed up metrics.

Reply:
this approach loses thread-safety, because during this line another thread could be somewhere between lines 51 and 55 for the same round ID.
Instead, we could change `putIfAbsent` to `computeIfAbsent` if your goal is to avoid double-creating `new GameRound()` objects that don't get used